code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -347,8 +352,59 @@
 	modes = list(MODE_CREW)
 	var/target = null //for targeting in processing
 	var/target_set = FALSE //have we set a target at any point?
+	///Var to track the linked detective gun
+	var/linked_gun_UID
+
+/obj/item/pinpointer/crew/Initialize(mapload)
+	. = ..()
+	var/obj/item/gun/energy/detective/gun_to_link = locate(/obj/item/gun/energy/detective) in get_turf(src)
+	if(gun_to_link)
+		link_gun(gun_to_link.UID())
```

Comment:
ok, instead of doing this, you could make a unique subtype of the pinpointer that only has the `MODE_DET` `mode` and then move the `attackby` logic to that, and remove this initialize override.

it feels weird to do this then include the option for someone to link it manually by using the two items on eachother. and if the concern is "I don't want to give the det a roundstart crew pinpointer" make a subtype.

Reply:
There was no concern truthfully, this seemed the easiest way to do it.
I want the functionality to be available to all pinpointers so it was easy to replace should one be lost. This was just my way of having the one issued to the det link without them having to do so manually.